api/controllers/game/resolve-five.js

```diff
@@ -0,0 +1,87 @@
+const { getCardName } = require('../../../utils/game-utils');
+module.exports = async function (req, res) {
+  
+  try {
+    const promiseGame = gameService.findGame({ gameId: req.session.game });
+    const promisePlayer = userService.findUser({ userId: req.session.usr });
+    const promiseCard = req.body.cardId ? cardService.findCard({ cardId: req.body.cardId }) : null;
+    const [game, player, cardToDiscard] = await Promise.all([promiseGame, promisePlayer, promiseCard]);
+
+    if (!game.topCard) {
+      throw new Error({ message: 'Cannot resolve 5 one-off with an empty deck' });
+    }
+
+    const gameUpdates = {
+      passes: 0,
+      turn: game.turn + 1,
+      resolving: null,
+      lastEvent: {
+        change: 'resolveFive',
+        discardedCards: cardToDiscard ? [cardToDiscard.id] : null,
```

Comment:
But the server shouldn't trust the client that those checks have already been made because anyone can forcibly make whatever request they want to the server. So each endpoint should validate itself that it's a fully legal move

Reply:
> But the server shouldn't trust the client that those checks have already been made because anyone can forcibly make whatever request they want to the server. So each endpoint should validate itself that it's a fully legal move

Copy, that makes sense

Reply:
> But the server shouldn't trust the client that those checks have already been made because anyone can forcibly make whatever request they want to the server. So each endpoint should validate itself that it's a fully legal move

Oops this was meant for the above comment about errors